BusBoard/Program.cs

```diff
@@ -16,31 +16,84 @@ static async Task Main(string[] args)
             .AddUserSecrets<Program>()
             .Build();
 
-        Console.WriteLine("Enter Stop Code:");
-        string id = Console.ReadLine()!;
-
-        // Build and Execute Request
+        
+        UserInputController userInput = new();
         TflAPIService tflAPI = new();
-
-        List<BusArrivalPrediction> nextFiveBusses;
+        PostcodeAPIService postcodeAPI = new();
 
         try
         {
-            nextFiveBusses = await tflAPI.GetNextNBussesAtStop(id, 5, config);
+            string postcode = userInput.GetPostcodeFromUser();
+
+            PostcodeData postcodeData = await postcodeAPI.GetPostcodeData(postcode);
+
+            if (postcodeData.Region != "London")
+            {
+                Console.WriteLine("Please Enter a London Postcode");
+                return;
+            }
+
+            Console.WriteLine("Searching...");
+            StopPointSearchResponse stopPointSearch = await tflAPI.GetStopPointsNearLocation(postcodeData.Latitude, postcodeData.Longitude);
+
+            if (stopPointSearch.StopPoints.Count < 2)
+            {
+                Console.WriteLine("Searching...");
+                stopPointSearch = await tflAPI.GetStopPointsNearLocation(postcodeData.Latitude, postcodeData.Longitude, true);

```

Comment:
What does `true` mean? It's not clear without looking at the implementation of the function

Reply:
Is this a place where a comment above the function call would be the thing to do?

Reply:
I think moreso I'd be looking at avoiding passing in a boolean at all - there's a few reasons for this
- readability - what does `true` mean
- is this function actually doing two things (and therefore should be 2 functions)

the latter point is a bit more of a marginal call in this case but you could resolve this by:
- using named parameters
```suggestion
                stopPointSearch = await tflAPI.GetStopPointsNearLocation(
                latitude: postcodeData.Latitude,
                longitude: postcodeData.Longitude, 
                useIncreasedSearchRadius: true);
```
- passing in a number
```suggestion
                stopPointSearch = await tflAPI.GetStopPointsNearLocation(
                latitude: postcodeData.Latitude,
                longitude: postcodeData.Longitude, 
                searchRadiusMiles: 5);
```
- creating two methods, with most of the details commonised (potentially private methods that look the same as above)